### PR TITLE
API Catchup to 3.3.0 version

### LINKED ISSUE #58: Update Rust wrapper to reflect 3.3.0 release

### PR DESCRIPTION
**Pull request by 9prady9**
Fixes #58 

More detailed information on the new release can be found at the
following URL

https://github.com/arrayfire/arrayfire/releases/tag/v3.3.0

### CODE REVIEW

**Comment by 9prady9** [comment]
@jramapuram Can you please review it. 

Please don't merge it yet, there is a minor commit left over but it is not related to 3.3.0 release. I will push it once i am done with it. Will let you know once i am done.

File: src/defines.rs [hunk 1]
```diff
@@ -51,22 +53,22 @@ pub enum AfError {
...
     fn fmt(&self, f: &mut Formatter) -> Result<(), FmtError> {
         let text = match *self {
-            Backend::AF_BACKEND_OPENCL  => "OpenCL",
-            Backend::AF_BACKEND_CUDA    => "Cuda",
-            Backend::AF_BACKEND_CPU     => "CPU",
-            Backend::AF_BACKEND_DEFAULT => "Default",
+            Backend::OPENCL  => "OpenCL",
+            Backend::CUDA    => "Cuda",
+            Backend::CPU     => "CPU",
+            Backend::DEFAULT => "Default",
```
Comment (jramapuram) [inline src/defines.rs]: Much cleaner. Good call

File: src/array.rs [hunk 2]
```diff
@@ -122,21 +137,68 @@ impl Array {
...
             let mut ret_val: i32 = 0;
-            af_get_backend_id(&mut ret_val as *mut c_int, self.handle as AfArray);
-            match ret_val {
-                1 => Backend::AF_BACKEND_CPU,
-                2 => Backend::AF_BACKEND_CUDA,
-                3 => Backend::AF_BACKEND_OPENCL,
-                _ => Backend::AF_BACKEND_DEFAULT,
+            let err_val = af_get_backend_id(&mut ret_val as *mut c_int, self.handle as AfArray);
+            match (err_val, ret_val) {
+                (0, 1) => Ok(Backend::CPU),
```
Comment (jramapuram) [inline src/array.rs]: Good way to squash the error check in. :+1:

**Comment by jramapuram** [comment]
Looks great! Great work!

**Comment by 9prady9** [comment]
@jramapuram I am done with the changes, you can review and merge if everything looks fine.

**Comment by 9prady9** [comment]
@jramapuram I am done with the changes, you can review and merge if everything looks fine.

**Comment by jramapuram** [comment]
Like the changes for the types. Will let you know if there are any issues. Merging into develop!

**Pull request merged**